concent_api/verifier/decorators.py

```diff
@@ -24,3 +28,38 @@ def wrapper(*args, **kwargs):
             # Remove any files left in VERIFIER_STORAGE_PATH.
             clean_directory(settings.VERIFIER_STORAGE_PATH)
     return wrapper
+
+
+def ensure_retry_of_locked_calls(func):
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        repeat_handler = RepeatHandler()
```

Comment:
I think this is not a good use case for a decorator. A decorator covers the whole function and that scope is too broad. And if it fails, you retry the whole function, not just the piece of code that obtains a lock.

It will be a source of bugs because doing this requires you to be extra careful when writing the function. You have to ensure that it can be retried without permanent side-effects. That's not an easy thing to do in a generic way.

There's also a possibility of easy mistakes with decorator order. If you use it **after** you start a transaction (using `@atomic`) database changes made before acquiring the lock will not be rolled back.

If you really want to have a reusable wrapper that manages a lock, you should make it a context manager. See https://jeffknupp.com/blog/2016/03/07/python-with-context-managers/. You could make a context manager that gets a query set as a parameter and performs a select query with a lock.